src/tests/tribol_proximity_check.cpp

```diff
@@ -0,0 +1,274 @@
+   */
+  double max_force_;
+
+  /**
+   * @brief Binning methods to test for each contact problem.
+   */
+  std::array<tribol::BinningMethod, 3> binning_methods_{ tribol::BINNING_CARTESIAN_PRODUCT, tribol::BINNING_GRID,
+                                                         tribol::BINNING_BVH };
+
+  void UpdateTribol( shared::ParMeshBuilder& mesh, int coupling_scheme_id )
+  {
+    tribol::updateMfemParallelDecomposition();
+    constexpr int cycle = 0;
+    constexpr tribol::RealT t = 0.0;
+    tribol::RealT dt = 1.0;
+    tribol::update( cycle, t, dt );
+
+    mfem::LinearForm r( &mesh.getNodesFESpace() );
+    r = 0.0;
+    tribol::getMfemResponse( coupling_scheme_id, r );
+    max_force_ = r.Max();
+
+    r.Print();
+
+    MPI_Allreduce( MPI_IN_PLACE, &max_force_, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD );
+  }
+
+  void SetUpCommonPlaneProblem( shared::ParMeshBuilder& mesh, const std::set<int>& contact_surf_1,
+                                const std::set<int>& contact_surf_2, tribol::RealT penalty,
+                                tribol::RealT binning_proximity, tribol::BinningMethod binning_method )
+  {
+    // set up tribol
+    constexpr int coupling_scheme_id = 0;
+    constexpr int mesh1_id = 0;
+    constexpr int mesh2_id = 1;
+    tribol::registerMfemCouplingScheme( coupling_scheme_id, mesh1_id, mesh2_id, mesh, mesh.getNodes(), contact_surf_1,
+                                        contact_surf_2, tribol::SURFACE_TO_SURFACE, tribol::NO_CASE,
+                                        tribol::COMMON_PLANE, tribol::FRICTIONLESS, tribol::PENALTY, binning_method );
+    tribol::setMfemKinematicConstantPenalty( coupling_scheme_id, penalty, penalty );
+    tribol::setBinningProximityScale( coupling_scheme_id, binning_proximity );
+
+    UpdateTribol( mesh, coupling_scheme_id );
+  }
+
+  void SetUp2DCommonPlaneProblem( tribol::BinningMethod binning_method )
+  {
+    constexpr int dim = 2;
+    // polynomial order of the finite element discretization
+    auto order = std::get<0>( GetParam() );
+    // binning proximity value
+    auto binning_proximity = std::get<1>( GetParam() );
+    // mesh interpenetration
+    auto mesh_interpenetration = std::get<2>( GetParam() );
+
+    // fixed options
+
+    // kinematic constant penalty stiffness
+    constexpr tribol::RealT penalty = 1.0;
+    // boundary element attributes of contact surface 1, the top surface of the bottom block
+    auto contact_surf_1 = std::set<int>( { 3 } );
+    // boundary element attributes of contact surface 2, the bottom surface of the top block
+    auto contact_surf_2 = std::set<int>( { 5 } );
+
+    // create a mesh with two blocks
+    // clang-format off
+    shared::ParMeshBuilder mesh( MPI_COMM_WORLD, shared::MeshBuilder::Merged( {
+      shared::MeshBuilder::SquareMesh( 1, 1 ),
+      shared::MeshBuilder::SquareMesh( 1, 1 )
+        .translate( { 0.0, 1.0 - mesh_interpenetration } )
+        .updateAttrib( 1, 2 )
```

Comment:
Not super clear how the old attributes got assigned and where the new ones come from. Looks like surface 1 has boundary attribute 3 and surface 2 has 5. Not sure where 6 comes from, for instance. Perhaps clarify so someone could pattern match if they needed to write their own test using this test harness.

Reply:
That's what the print boundary attributes method is for, so you know where the boundary attributes are.  I don't see any documentation in mfem's mesh creation methods, so I don't think we should add any notes etc. here because it isn't defined in mfem's documentation what the attribute sets are, so they could change.  This is why I think picking attributes based on a bounding box might be more useful and general.